File: test/end-to-end/Graphs/TypeScript/Data/UnweightedNode.ts
import { INode } from "./INode";

/**
 * An unweighted, undirected node in a graph.
 */
export class UnweightedNode<T> implements INode<UnweightedNode<T>> {
    public data: T;
    private neighborNodes: UnweightedNode<T>[];

    public constructor(data: T) {
        this.data = data;
        this.neighborNodes = [];
    }

    public addNeighbor(node: UnweightedNode<T>): void {
        this.neighborNodes.push(node);
    }

    public getNeighborsInOrder(): UnweightedNode<T>[] {
        return this.neighborNodes;
    }
}

File: test/end-to-end/Graphs/C#/Data/UnweightedNode.cs
using Graphs.Data;
using System.Collections.Generic;

namespace Graphs.Data
{

    /// <summary>
    /// An unweighted, undirected node in a graph.
    /// </summary>
    public class UnweightedNode<T> : INode<UnweightedNode<T>>
    {
        public T Data;
        private List<UnweightedNode<T>> neighborNodes;

        public UnweightedNode(T data)
        {
            this.Data = data;
            this.neighborNodes = new List<UnweightedNode<T>>();
        }

        public void AddNeighbor(UnweightedNode<T> node)
        {
            this.neighborNodes.Add(node);
        }

        public List<UnweightedNode<T>> GetNeighborsInOrder()
        {
            return this.neighborNodes;
        }
    }
}

